cursive-core/src/buffer.rs

```diff
@@ -0,0 +1,322 @@
+//! Output buffer
+
+use crate::backend::Backend;
+use crate::theme::ConcreteStyle;
+use crate::Vec2;
+
+use unicode_segmentation::UnicodeSegmentation;
+use unicode_width::UnicodeWidthStr;
+
+/// The width of a cell.
+///
+/// Most characters are single-width. Some asian characters and emojis are double-width.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+enum CellWidth {
+    Single,
+    Double,
+}
+
+impl Default for CellWidth {
+    fn default() -> Self {
+        CellWidth::Single
+    }
+}
+
+impl CellWidth {
+    pub fn from_usize(width: usize) -> Self {
+        match width {
+            1 => CellWidth::Single,
+            2 => CellWidth::Double,
+            _ => panic!("expected width of 1 or 2 only."),
```

Comment:
Can that be this:
```diff
+            n => panic!("expected width of 1 or 2 only, got {n}."),
```

It shows `got 0.` in case of `\0`(nuls) in string.

Reply:
Thanks!

Indeed, I'm now skipping zero-width graphemes entirely.